### PR TITLE
[Docs] - Created a documentation website for flutter_deck with static_shock (Resolves #93)

### LINKED ISSUE #93: docs: Create a guide website
**Description**

Create a guide website for easy reading and navigating of common goals, as well as documentation of the intended architecture and extension points.

**Tooling**

I can get this started with [static_shock](https://staticshock.io) and write some initial guides.

### PR DESCRIPTION
**Pull request by matthew-carroll**
[Docs] - Created a documentation website for flutter_deck with static_shock (Resolves #93)

Located in the repo at `/doc/website`.

I haven't added a GitHub Pages deployment Action. I can do that next. But before that, @mkobuolys you need to activate GitHub Pages for this repo. When you do, you'll be able to choose between deploy from a branch vs GitHub Actions. You'll want to choose GitHub Actions. Then, I can configure a workflow to build and deploy.

To get the links working on this website when it's deployed to GitHub Pages, I need to add some URL logic to Static Shock and update this website. This is because GitHub pages injects an extra path segment into every URL.

@mkobuolys I recommend pulling down this PR and then doing the following:
 * Run `dart pub global activate static_shock_cli`.
 * Run `dart pub get` in the website project.
 * Run `shock build` to make sure there aren't any build errors for you.
 * Run `shock serve` to explore the website locally.

Screenshot 2024-07-27 at 12 54 39 PM
Screenshot 2024-07-27 at 12 54 54 PM

### CODE REVIEW

File: doc/website/source/get-started.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  ),
+  <...>
+);
+```
+
+## Next Steps
+You have a Flutter project with a configured `FlutterDeckApp` widget. Consider exploring the
+following guides to build up your slide show:
+
+ *
```
Comment (mkobuolys) [inline doc/website/source/get-started.md:126]: The list is empty, should we add something here?

File: doc/website/source/guides/code-generation.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+title: Code Generation
+navOrder: 2
+---
+This package comes with a [mason][https://pub.dev/packages/mason] template that can be used to generate a new slide for the slide deck.
```
Comment (mkobuolys) [inline doc/website/source/guides/code-generation.md:5]:
```suggestion
This package comes with a [mason](https://pub.dev/packages/mason) template that can be used to generate a new slide for the slide deck.
```

File: doc/website/source/guides/code-generation.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Ensure you have the [mason_cli][https://pub.dev/packages/mason_cli] installed:
```
Comment (mkobuolys) [inline doc/website/source/guides/code-generation.md:7]:
```suggestion
Ensure you have the [mason_cli](https://pub.dev/packages/mason_cli) installed:
```

File: doc/website/source/guides/code-generation.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+```sh
+dart pub global activate mason_cli
+```
+
+Install the [flutter_deck_slide][https://brickhub.dev/bricks/flutter_deck_slide] template:
```
Comment (mkobuolys) [inline doc/website/source/guides/code-generation.md:13]:
```suggestion
Install the [flutter_deck_slide](https://brickhub.dev/bricks/flutter_deck_slide) template:
```

File: doc/website/source/guides/localization.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
+title: Localization
+navOrder: 7
+---
+This package comes with a built-in localization support. You can change the locale of the slide deck at runtime (see [controls](#controls)). The updated locale will be applied to the whole slide deck.
```
Comment (mkobuolys) [inline doc/website/source/guides/localization.md:5]: The #controls link does not guide anywhere.

File: doc/website/source/guides/widgets.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        ),
+      ),
+    );
+  }
+}
+```
+
+![FlutterDeckCodeHighlight](https://github.com/mkobuolys/flutter_deck/blob/main/images/code_highlighting.png?raw=true)
+
+## 📦 Accessing slide deck state from the code
```
Comment (mkobuolys) [inline doc/website/source/guides/widgets.md:95]: This section should probably have a dedicated page under Guides.

**Review by mkobuolys — CHANGES_REQUESTED** [review]
Thanks for the documentation website, that's awesome! About the website deployment, I already have a [domain](https://flutterdeck.dev/), does that change anything? I can probably take care of the deployment part myself once this PR gets merged.

File: doc/website/source/guides/widgets.md (hunk 6 above)
Comment (matthew-carroll) [reply doc/website/source/guides/widgets.md:95]: Should it? Seems like a small amount of info that's highly related to widgets. I'll move it if you'd like, but wanted to double check.

It's also possible that "widgets" isn't the best organizing concept in the first place. We could also take pieces from this file and move them to other guides. Let me know what's best.

**Comment by matthew-carroll** [comment]
@mkobuolys I resolved all issues except for the last one. I followed up with you in that comment.

As for deployments - yes you can use a custom domain. In fact, if you use a custom domain then you shouldn't need the extra feature that I'm planning to add to Static Shock related to base URLs. Instead, it should just work, the same way that staticshock.io works. 

Are you familiar with deploying with GitHub pages? I would still recommend using that approach, even with a custom domain, because you can set it up to auto-build and deploy every time somebody merges to main. So you never have to worry about a separate path to update docs, which reduces friction, which increases the likelihood that people actually update the docs.

**Comment by matthew-carroll** [comment]
Hey @mkobuolys - any followup?

**Review by mkobuolys — APPROVED** [review]
Hey, sorry for the late reply and thanks for you effort! I will take care of the deployment to GH pages (in fact, flutter_deck already uses them for the example presentation). Looking forward to checking the static_shock myself and improving the docs!

**Pull request merged**